webview-ui/src/components/Message.tsx

```diff
@@ -0,0 +1,144 @@
+import { VSCodeButton, VSCodeDivider } from "@vscode/webview-ui-toolkit/react";
+import Markdown from "react-markdown";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import { darcula } from "react-syntax-highlighter/dist/esm/styles/hljs";
+import { ChatRole, CodeBlock, MessageBlock } from "../utilities/ChatModel";
+import { ExtensionMessenger } from "../utilities/ExtensionMessenger";
+import { ImagePaths } from "../types";
+
+interface MessageProps {
+  role: ChatRole;
+  messageBlocks: MessageBlock[];
+  extensionMessenger: ExtensionMessenger;
+  imagePaths?: ImagePaths;
+}
+export const Message: React.FC<MessageProps> = ({
+  role,
+  messageBlocks,
+  extensionMessenger,
+  imagePaths,
+}) => (
+  <div>
+    <VSCodeDivider role="separator" />
+    <div className="message">
+      <MessageHeader role={role} assistantIconUri={imagePaths?.byoladIconUri} />
+      <div className="message-block">
+        {messageBlocks.map((messageBlock) => {
+          if (messageBlock.type === "text") {
+            return <TextMessageBlock>{messageBlock.content}</TextMessageBlock>;
+          } else if (messageBlock.type === "code") {
+            return (
+              <CodeMessageBlock
+                languageId={(messageBlock as CodeBlock).languageId}
+                extensionMessenger={extensionMessenger}
+              >
+                {messageBlock.content}
+              </CodeMessageBlock>
+            );
+          } else {
+            throw new Error("Unknown message block type");
```

Comment:
I think that this would be better handled by just doing `return <></>`. We don't want to crash our entire React app just because of an unknown message block type. It would just be better to not render it and render the rest of the stuff.

Reply:
Good point, good point